### DISCUSSION TITLE
Using Spektral for RS

### DISCUSSION THREAD

**Discussion by vicbeldo98** [danielegrattarola/spektral#387] [General]
Hello! I am a student trying to use Spektral for building a recommendation system based on GNN, so my task would be link prediction.
Particularly, I want to adapt it to the Yelp Dataset (users rate business).
I have a question that might be very easy to answer. Can you specify nodes with different initial features? I have two types of nodes: business and users. I have been looking through the documentation but I do not find any way.

Thank you in advanced for the help :)

**Comment by danielegrattarola** [comment]
Hey Victoria, 

unfortunately Spektral does not support heterogeneous features natively, you have to deal with them yourself. 
It can be as easy as having two separate linear projections applied to nodes of either type, or you can do it as a pre-processing step and make sure that both types have the same feature dimensions. 

All the best
Daniele
